### ISSUE TITLE
New git version is not found

### ISSUE THREAD

**Issue by vorner** [nabijaczleweli/cargo-update#78]
It seems `cargo install-update --all --git` doesn't pick newest git versions from the repository. Snippet from when I run it:

```
Package                Installed                                 Latest                                    Needs update
alacritty              2466f81d5e0cb56dd00231f63f2dbcef1840196a  2466f81d5e0cb56dd00231f63f2dbcef1840196a  No
```

However, current alacritty commit is `3d75c491912952d2a18fb34ab81221b1bded07d5`.

**Comment by nabijaczleweli** [comment]
What's your `.crates.toml`?

**Comment by vorner** [comment]
```
[v1]
"alacritty 0.1.0 (git+https://github.com/jwilm/alacritty#2466f81d5e0cb56dd00231f63f2dbcef1840196a)" = ["alacritty"]
"bigbuffer 0.1.0 (git+https://github.com/vorner/bigbuffer#40cb0173c142c70c695a8df73a4c2564f0759b5e)" = ["bigbuffer"]
"cargo-bundle 0.2.0 (git+https://github.com/burtonageo/cargo-bundle#1627e40494178314c00db1692c0f95aa7b1e2e1d)" = ["cargo-bundle"]
"cargo-check 0.2.2 (registry+https://github.com/rust-lang/crates.io-index)" = ["cargo-check"]
"cargo-deb 1.9.0 (registry+https://github.com/rust-lang/crates.io-index)" = ["cargo-deb"]
"cargo-expand 0.3.12 (registry+https://github.com/rust-lang/crates.io-index)" = ["cargo-expand"]
"cargo-lichking 0.5.4 (registry+https://github.com/rust-lang/crates.io-index)" = ["cargo-lichking"]
"cargo-outdated 0.7.0 (registry+https://github.com/rust-lang/crates.io-index)" = ["cargo-outdated"]
"cargo-tree 0.17.0 (registry+https://github.com/rust-lang/crates.io-index)" = ["cargo-tree"]
"cargo-update 1.5.0 (registry+https://github.com/rust-lang/crates.io-index)" = ["cargo-install-update", "cargo-install-update-config"]
"cargo-watch 6.0.0 (registry+https://github.com/rust-lang/crates.io-index)" = ["cargo-watch"]
"clippy 0.0.193 (registry+https://github.com/rust-lang/crates.io-index)" = ["cargo-clippy", "clippy-driver"]
"compile-commands-post 0.1.0 (git+https://github.com/vorner/compile-commands-post#0089189a81abfea26c2d58357ffddacb215e5f6f)" = ["compile-commands-post"]
"cross 0.1.14 (registry+https://github.com/rust-lang/crates.io-index)" = ["cross"]
"groupby 0.1.3 (registry+https://github.com/rust-lang/crates.io-index)" = ["groupby"]
"jsonrpc-param-ord 0.1.0 (git+https://github.com/vorner/jsonrpc-param-ord#2559b5e0a90d12329c3c5c20a0ff672ce4aaf4fb)" = ["jsonrpc-param-ord"]
"pgz 0.1.0 (git+https://github.com/vorner/pgz#f73f3c3b7d48abc77b63a8d9d564d48b3f05a808)" = ["pgz"]
"racer 2.0.13 (registry+https://github.com/rust-lang/crates.io-index)" = ["racer"]
"ripgrep 0.8.1 (registry+https://github.com/rust-lang/crates.io-index)" = ["rg"]
"runner 0.3.4 (registry+https://github.com/rust-lang/crates.io-index)" = ["runner"]
"skim 0.3.2 (registry+https://github.com/rust-lang/crates.io-index)" = ["sk"]
"svgbob_cli 0.2.7 (registry+https://github.com/rust-lang/crates.io-index)" = ["svgbob"]
"watchexec 1.8.6 (registry+https://github.com/rust-lang/crates.io-index)" = ["watchexec"]
"xargo 0.3.12 (registry+https://github.com/rust-lang/crates.io-index)" = ["xargo"]
```

(yes, the version matches the one shown above)

**Comment by nabijaczleweli** [comment]
Interesting, I can't reproduce that 🤔

Try removing `$TMP/cargo-update` and re-run `cargo-update`? That's the most obvious failure point I can see.

**Comment by vorner** [comment]
That might be related. I did a restart in between (my tmp is a ramdisk, so it wipes it) and it did find the new version now. Does it keep some git snapshot there, or something?

**Comment by nabijaczleweli** [comment]
Yep, and apparently we mishandle that somehow. Thanks for confirming!

**Comment by nabijaczleweli** [comment]
Released in [`v1.5.1`](https://github.com/nabijaczleweli/cargo-update/releases/tag/v1.5.1).
